frb_rust/src/handler.rs

```diff
@@ -122,38 +122,28 @@ impl<E: Executor, EH: ErrorHandler> Handler for SimpleHandler<E, EH> {
         });
     }
 
-    fn wrap_sync<SyncTaskFn>(
+    fn wrap_sync<SyncTaskFn, TaskRet>(
         &self,
         wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
     ) -> WireSyncReturnStruct
     where
-        SyncTaskFn: FnOnce() -> Result<SyncReturn<Vec<u8>>> + UnwindSafe,
+        WireSyncReturnStruct: From<TaskRet>,
```

Comment:
Well, I am thinking about whether we can avoid so many usages of `WireSyncReturnStruct`. Maybe we can do this:

* Create a new struct, say (very naive naming), `struct WireSyncReturnData(Vec<u8>);`
* `impl From<WireSyncReturnData> for WireSyncReturnStruct { ... }`. Notice this will be the *only* place where we write down the dangerous `into_leak_vec_ptr` and raw pointers!
* Remove all other `impl From<...> for WireSyncReturnStruct`. Instead, convert any type into a `WireSyncReturnData`. That would be much safer, since we are just converting things to `Vec<u8>` which is perfectly safe.
* The bounds here may become `WireSyncReturnData: From<TaskRet>`

Reply:
Agreed, `into_leak_vec_ptr` should be wrapped up.